redbeams/src/main/java/com/sequenceiq/redbeams/service/UserGeneratorService.java

```diff
@@ -13,14 +17,34 @@
 
     private static final int USER_NAME_LENGTH = 10;
 
+    private static final String UUID_EXTRACTOR_REGEXP = "(\\w{8})-(\\w{4})-\\w(\\w{3})-\\w(\\w{3})-(\\w{12})";
+
+    private static final int MAX_LENGTH_PASSWORD = 30;
+
+    @Inject
+    private UUIDGeneratorService uuidGeneratorService;
+
     public String generateUserName() {
         return ThreadLocalRandom.current().ints(0, MAX_RANDOM_INT_FOR_CHARACTER)
                 .limit(USER_NAME_LENGTH).boxed()
                 .map(i -> Character.toString((char) ('a' + i)))
                 .collect(Collectors.joining());
     }
 
+    /*
+    Generates a password using a UUID.
+    Provider specific limitations:
+    - AWS: password length in cf template can be 30 characters long at most
+     */
     public String generatePassword() {
-        return UUID.randomUUID().toString();
+        Pattern pattern = Pattern.compile(UUID_EXTRACTOR_REGEXP);
+        String uuid = uuidGeneratorService.randomUuid();
+        Matcher matcher = pattern.matcher(uuid);
+        if (!matcher.matches()) {
+            return uuid.substring(0, MAX_LENGTH_PASSWORD < uuid.length() ? MAX_LENGTH_PASSWORD - 1 : uuid.length());
```

Comment:
(If this remains) Don't subtract one from MAX_LENGTH_PASSWORD, or else the resulting string will be MAX_LENGTH_PASSWORD - 1 characters in length.